### PR TITLE
[CPU] Fix Loop output shape mismatch on zero iterations

### PR DESCRIPTION
**Pull request by liubo-intel**
### Details:
 - *Fixes a shape mismatch crash when a Loop (opset5) executes zero
  iterations: a loop-carried output (a body output fed back via a
  merged input) used to fall back to the body's never-executed output
  memory and get its shape nullified to [0,0,0], breaking downstream
  Eltwise-like shape inference. It now forwards the shape and data of
  the merged input's initial value instead, mirroring the equivalent
  GPU plugin fix.*
 - *Covers both the static (`execute()`) and dynamic
  (`executeDynamicImpl()`/`reshapeAndFillOutput()`) shape paths.*
 - *Adds `LoopZeroIterationsCPUTest` covering trip_count=0 and
  false execution_condition, with a manually computed reference*

### Tickets:
 - *CVS-191389*

### CODE REVIEW

**Comment by maxnick** [comment]
@nshchego , could you please review?

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp [hunk 1]
```diff
@@ -710,7 +713,9 @@ void TensorIterator::execute(const dnnl::stream& strm) {
     }
 
     for (auto& mapper : last_mappers) {
-        mapper->execute(strm, -1);
+        // If the body has produced nothing, a loop carried dependency output keeps its initial value
+        const bool useInitialValue = !has_executed && mapper.zeroIterMapper;
```
Comment (nshchego) [inline src/plugins/intel_cpu/src/nodes/tensoriterator.cpp]: `use_initial_value` - could you use snake case here and below?

**Review by nshchego — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp (hunk 1 above)
Comment (liubo-intel) [reply src/plugins/intel_cpu/src/nodes/tensoriterator.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp [hunk 2]
```diff
@@ -699,6 +701,7 @@ void TensorIterator::execute(const dnnl::stream& strm) {
         }
 
         sub_graph.Infer();
+        has_executed = true;
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/tensoriterator.cpp]: In fact we can hoist `has_executed` out of the loop body and the condition is 
`const bool has_executed = (max_num_iter != 0)  && continue_cond`

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp [hunk 3]
```diff
@@ -736,6 +742,7 @@ void TensorIterator::executeDynamicImpl(const dnnl::stream& strm) {
...
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/tensoriterator.cpp]: The same thing, the `has_executed ` may be hoisted out of the loop.

File: src/plugins/intel_cpu/src/nodes/tensoriterator.h [hunk 4]
```diff
@@ -191,10 +196,22 @@ class TensorIterator : public Node {
...
-        before_mappers,                                        /// < Applied before each iteration
-        after_mappers,                                         /// < Applied after each iteration
-        back_mappers;                                          /// < Applied before each iteration for dynamic shapes
+    /**
+     * Pair of helpers to fill one node output port when the loop is over.
+     * 'mapper' takes the value produced by the body on the last executed iteration, while
+     * 'zeroIterMapper' takes the initial value of the corresponding loop carried dependency
+     * (merged input) and is used when the body has not been executed at all. It stays empty
+     * for the output ports which are not loop carried dependencies.
+     */
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/tensoriterator.h]: Please condensate the comment to two lines.

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp [hunk 5]
```diff
@@ -893,9 +910,67 @@ void TensorIterator::reshapeSubgraphInput() {
...
+        return -1;
+    }
+
+    return input_rule->from;
+}
+
+// In case the body has not been executed at all (zero trip count or false initial execution condition)
+// the value of a loop carried dependency is still its initial value, so the output has to be filled from
+// the corresponding node input directly. The body output memory cannot be used here: it has neither a
+// valid shape (it is left undefined, which used to be nullified to zeros) nor valid data.
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/tensoriterator.cpp]: Please condensate the comment to two lines.

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/loop.cpp [hunk 6]
```diff
@@ -454,6 +455,105 @@ class LoopZeroDimBackEdgeCPUTest : public SubgraphBaseTest {
...
+        // the body is executed only if both the trip count and the execution condition allow it
+        executed_iterations = exec_cond ? trip_count : 0;
+
+        InputShape input_shape = {{-1, -1, -1}, {{2, 3, 4}, {1, 5, 5}}};  // infer more than once
+        InputShape exec_cond_shape = {{1}, {{1}, {1}}};
+        init_input_shapes({input_shape, exec_cond_shape});
+
+        auto input = std::make_shared<ov::op::v0::Parameter>(netType, inputDynamicShapes[0]);
+        // the execution condition is a parameter to keep the loop output shape dynamic
+        auto exec_condition = std::make_shared<ov::op::v0::Parameter>(ov::element::boolean, inputDynamicShapes[1]);
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/loop.cpp]: This always forces dynamic path, but it's crucial to also test the static `execute()` path.
Please add a test case with zero trip count and a const exec_condition set to true plus static input shapes,  so the node stays static and drives `last_mappers / zeroIterMapper`. Assert output data equals the initial value (not just the shape).

**Review by maxnick — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp (hunk 2 above)
Comment (liubo-intel) [reply src/plugins/intel_cpu/src/nodes/tensoriterator.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/tensoriterator.h (hunk 4 above)
Comment (liubo-intel) [reply src/plugins/intel_cpu/src/nodes/tensoriterator.h]: Done

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/loop.cpp (hunk 6 above)
Comment (liubo-intel) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/loop.cpp]: static path testcase added.

**Review by maxnick — APPROVED** [review]
